### PR TITLE
Test similar torrent

### PR DESCRIPTION
**Pull request by joriscarrier**
#3808

### CODE REVIEW

**Comment by arvidn** [comment]
If you want to use a different test framework, I think all tests should be transitioned over, and I don't think such change should be conflated with adding a test

**Comment by arvidn** [comment]
What are the benefits of `pytest` over the unit test framework currently used?

File: bindings/python/tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+from uuid import uuid4
+import os
+
+import pytest
+import libtorrent as lt
+
+
+@pytest.yield_fixture
+def torrent_info(tmpdir):
```
Comment (arvidn) [inline bindings/python/tests/conftest.py]: what does this function do? it doesn't look like it's testing anything.

File: bindings/python/tests/test_torrent.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+                )
+            create_torrent.set_file_hash(
+                file_indice * torrent_indice,
+                _file.filehash.to_bytes()
+            )
+
+    entry = create_torrent.generate()
+    ti = lt.torrent_info(entry)
+
+    assert ti.name() == 'similar_torrent'
```
Comment (arvidn) [inline bindings/python/tests/test_torrent.py]: this test doesn't really seem to capture the logic that's being exercised here. surely you would need to ensure that the torrent has a field for similar torrents, and maybe some properties of that field, like which info-hashes are in there and no duplicates etc.

File: bindings/python/tests/test_torrent.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    create_torrent = lt.create_torrent(file_storage)
+
+    for torrent_indice, ti in enumerate(torrents_info):
+        create_torrent.add_similar_torrent(ti.info_hash())
+
+        for file_indice, _file in enumerate(ti.files()):
+            for piece_indice in range(ti.num_pieces()):
+                create_torrent.set_hash(
+                    piece_indice * torrent_indice,
+                    ti.hash_for_piece(piece_indice)
```
Comment (arvidn) [inline bindings/python/tests/test_torrent.py]: this line is non-sensical. you're looking over the number of pieces of *some other* torrent, and assigning it to the torrent you're creating.

File: bindings/python/tests/test_torrent.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                )
+            create_torrent.set_file_hash(
+                file_indice * torrent_indice,
```
Comment (arvidn) [inline bindings/python/tests/test_torrent.py]: this line is also non-sensical. you're looping over the files of *some other* torrent, and assigning to the one you're creating

File: bindings/python/tests/test_torrent.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+    file_storage.set_name('similar_torrent')
+
+    create_torrent = lt.create_torrent(file_storage)
+
+    for torrent_indice, ti in enumerate(torrents_info):
+        create_torrent.add_similar_torrent(ti.info_hash())
+
+        for file_indice, _file in enumerate(ti.files()):
+            for piece_indice in range(ti.num_pieces()):
```
Comment (arvidn) [inline bindings/python/tests/test_torrent.py]: what's the point of looping over all pieces once per file? the pieces are the same every time

**Comment by arvidn** [comment]
it appears you have misunderstood how to create torrents and how "similar torrents" work. would you mind pointing out parts of the documentation that you find unclear, ambiguous, or hard to follow?

File: bindings/python/tests/conftest.py (hunk 1 above)
Comment (joriscarrier) [reply bindings/python/tests/conftest.py]: https://docs.pytest.org/en/latest/fixture.html
